### DISCUSSION TITLE
API architecture

### DISCUSSION THREAD

**Discussion by BrianHumlicek** [TheAxelander/OpenBudgeteer#214] [General]
Is there a reason the API is separate from the Blazor project?

Brian

**Comment by TheAxelander** [comment]
The API is only interacting with the backend/database and not with Blazor as Frontend. Mainly the idea was to make the API as container available separately and to make it independent from the "App Framework" which currently only Blazor but could be potentially extended in future with other frameworks like Avalonia, MAUI or anything else. 

Does this makes sense?

**Comment by BrianHumlicek** [comment]
Ok, I think I understand.  Stated differently, the API is potentially useful, but didn't clearly belong with the App Framework, so, it offered separately for consumption if someone needs it.

The reason I'm asking is because I have a bunch of data I want to initialize my database with, and was considering using the API to do it.  When I realized it wasn't part of the main container, I considered making a local change to include it, but I wanted to make sure there wasn't an unforeseen reason not to before I did.

In an effort to keep discussion on topic and organized, I'm going to start a new thread regarding the data import, and potential solutions.

Brian
